projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.native.js

```diff
@@ -63,20 +73,72 @@ export default function Player( { isSelected, attributes } ) {
 		poster,
 	} );
 
-	const { preview, isRequestingEmbedPreview } = usePreview( videoPressUrl );
-	const html = addTokenIntoIframeSource( preview?.html, token );
+	// Helper to invalidate the preview cache.
+	const invalidateResolution = useDispatch( coreStore ).invalidateResolution;
+	const invalidatePreview = useCallback( () => {
+		invalidateResolution( 'getEmbedPreview', [ videoPressUrl ] );
+	}, [ videoPressUrl, invalidateResolution ] );
+
+	const { preview = {}, isRequestingEmbedPreview } = usePreview( videoPressUrl );
+	const html = addTokenIntoIframeSource( preview.html, token );
+
+	const cancelPreviewCheckInterval = useCallback( () => {
+		clearInterval( previewCheckInterval.current );
+	}, [ previewCheckInterval ] );
+
+	// Fetch the preview until it's ready
+	useEffect( () => {
+		// Wait for the client player to load
+		if ( ! isPlayerLoaded ) {
+			return;
+		}
+		previewCheckInterval.current = setInterval( () => {
```

Comment:
I tested the case of loading a post with a VideoPress video and noticed that the interval is never canceled. As you can see in the video capture, `previewCheckAttempts` remains with value `1`, and `isRequestingEmbedPreview` is always `true` although the preview is rendered.


https://user-images.githubusercontent.com/14905380/232114692-b9cfc14b-08c7-4626-b6c5-1a17c9d12f93.mp4

Reply:
I checked the value of `isRequestingEmbedPreview` in the render function and it changes to `false` while the interval function still gets `true` 🤔. @jhnstn let me know if you can reproduce this issue, thanks!

Reply:
That's the same behavior I am seeing on Android but I haven't seen that on iOS.  I'll check again.

Reply:
What's odd about this is that the embed endpoint for VideoPress _should_ be returning a preview with height and width properties when opening a post with an existing video. I have not seen this before but good to know that this can happen.

I'm still not sure why `isRequestingEmbedPreview` getting stuck on true. I've tried changing the order of logic but nothing seems to change.

Reply:
> What's odd about this is that the embed endpoint for VideoPress should be returning a preview with height and width properties when opening a post with an existing video. I have not seen this before but good to know that this can happen.

Yep, I noticed in the response of the preview that the `provider_name` field is `Embed handler` which is the default response when it can't infer the provider. In fact, when the provider is not inferred is also when the width and height are empty. The second preview request responds `provider_name` with `VideoPress` and the proper dimensions. As I comment below, I think the fact that is not returning the dimensions is related to the effect and the hook's dependencies.

> I'm still not sure why `isRequestingEmbedPreview` getting stuck on true. I've tried changing the order of logic but nothing seems to change.

I think this can be related to using `setInterval` within the `useEffect` and the hook's dependencies. I played around with this part and seems we could replace it with `setTimeout`:

```patch
diff --git forkSrcPrefix/projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.native.js forkDstPrefix/projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.native.js
index 785230d0c32b6f497dd47431c7975511f20255c9..ebb1a3afcdc101c96a9ee9a91f82fa29357af766 100644
--- forkSrcPrefix/projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.native.js
+++ forkDstPrefix/projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.native.js
@@ -88,28 +88,21 @@ export default function Player( { isSelected, attributes } ) {
 
 	// Fetch the preview until it's ready
 	useEffect( () => {
-		// Wait for the client player to load
-		if ( ! isPlayerLoaded ) {
+		// Only check preview once it's available.
+		if ( ! isPlayerLoaded || isRequestingEmbedPreview ) {
 			return;
 		}
-		previewCheckInterval.current = setInterval( () => {
-			// if the preview is ready or we reached the max attempts, clear the interval
-			if ( isPreviewReady( preview ) || previewCheckAttempts > VIDEO_PREVIEW_ATTEMPTS_LIMIT ) {
-				return cancelPreviewCheckInterval();
-			}
-
-			// Avoid lapping any prior request
-			if ( isRequestingEmbedPreview ) {
-				return;
-			}
-
-			// Invalidate the cache and wait for the next interval since the preview is not ready yet
-			setPreviewCheckAttempts( previewCheckAttempts + 1 );
-			invalidatePreview();
-		}, 1000 );
+
+		// Request another preview if it's not ready and haven't exceeded request attemtps.
+		if ( ! isPreviewReady( preview ) && previewCheckAttempts <= VIDEO_PREVIEW_ATTEMPTS_LIMIT ) {
+			previewCheckInterval.current = setTimeout( () => {
+				invalidatePreview();
+				setPreviewCheckAttempts( previewCheckAttempts + 1 );
+			}, 1000 );
+		}
 
 		return cancelPreviewCheckInterval;
-	}, [ preview, isPlayerLoaded ] );
+	}, [ preview, isPlayerLoaded, isRequestingEmbedPreview, previewCheckAttempts ] );
 
 	const onSandboxMessage = useCallback( message => {
 		if ( message.event === 'videopress_loading_state' && message.state === 'loaded' ) {

```

Let me know if you'd like to proceed with this option.

Reply:
> That's the same behavior I am seeing on Android after uploading. But I've haven't seen that on iOS or on Android with existing videos.

I'm wondering if the problem on Android has to do with the caching layer on the `api-fetch` handler ([reference](https://github.com/WordPress/gutenberg/blob/7fa97000771188e2d1b885991e2332679ec6790e/packages/react-native-editor/src/api-fetch-setup.js#L19-L20)). On Android `GET` requests are cached in the host app by default unless it's specified, maybe for the previews we should disable it.